sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientTelemetryConfig.java

```diff
@@ -205,6 +204,62 @@ EnumSet<TagName> getMetricTagNames() {
         return this.metricTagNames;
     }
 
+    /**
+     * Sets the categories of metrics that should be emitted. By default the following categories will be enabled:
+     * OperationSummary, RequestSummary, DirectChannels, DirectRequests, System
+     * (the System and OperationSummary metrics are always collected and can't be disabled when enabling Cosmos metrics)
+     * For most use-cases that should be sufficient. An overview of the different metric categories can be found here:
+     * https://aka.ms/azure-cosmos-metrics
+     *
+     * @param categories - a comma-separated list of metric categories that should be emitted
+     * @return current CosmosClientTelemetryConfig
+     */
+    public CosmosClientTelemetryConfig metricCategories(String... categories) {
+        if (categories == null || categories.length == 0) {
+            this.metricCategories = MetricCategory.DEFAULT_CATEGORIES;
+        }
+
+        Map<String, MetricCategory> categoryMap = new HashMap<>();
+        for (MetricCategory level : MetricCategory.values()) {
+            categoryMap.put(level.toLowerCase(), level);
+        }
+
+        Stream<MetricCategory> categoryStream =
+            Arrays.stream(categories)
+                  .map(rawCategory -> rawCategory.toLowerCase(Locale.ROOT))
+                  .filter(category -> !Strings.isNullOrWhiteSpace(category))
```

Comment:
should the `filter` happen first, then `map`, else still a nullRef risk

Reply:
Fixed - good catch.